contracts/ProcedureTable.sol

```diff
@@ -450,7 +791,7 @@ library ProcedureTable {
 
         // The capabilities are parsed here. We neeed to pass in the Procedure
         // struct as solidity can't return complex data structures.
-        _parseCaps(p,caps);
+        // _parseCaps(p,caps);
 
         // If the keyIndex is not zero then that indicates that the procedure
         // already exists. In this case *WE HAVE NOT OVERWRITTEN * the values,
```

Comment:
So `ProcedureTable#insert` does not do re-insertion. So re-registration doesn't work in `Kernel#_registerAnyProcedure`

Reply:
Can you expand on this? I'm not quite sure what you mean.

Reply:
After some consideration it may make sense to prevent re-registration, so let's keep this for now.

Reply:
I'm just not sure how it prevents re-registration.

Reply:
Do you mean you can't "overwrite" a procedure without first deleting it? If so that's true.

Reply:
I'm just taking note that a procedure can't update/register an already existing procedure (including itself), unless it has both `REG_PROC` and `DELETE_PROC` caps (or `PROC_CALL`) beforehand.

Reply:
Correct. I can't remember if we discussed this or whether this is a similar issue that came up before. I think an important factor here might be that if I can re-register a procedure to be a "nothing/dummy" procedure, that's equivalent to deleting it from a security perspective. For that reason alone my first impression here is that we should keep this behaviour. What are you're thoughts?

Reply:
> > So at [Kernel.sol:704](https://github.com/Daolab/beakeros-evm1/pull/129/files#diff-b850c21d841e79dd4d4006531e138cfeR704), we're setting that `0` is not a valid procedureId. Just to make sure, are there any other reasons than preventing collisions during a "null" `currentProcedure`?
> 
> That was previously a requirement from our original tests. There's no functional reason we need to prevent it.

The issue that I see is that you could register a proc with an id `0`, which would make `#_seralizeCapArray` ignore it's c-list. I thought that logic is included simply for bootstrapping.

Reply:
Let's keep it for now, since it makes sense. Re-registration would be equivalent to deletion, which makes it more complicated.

Reply:
I see, yes. In a sense that means procedure == 0 kind of means "kernel". The currentProcedure == 0 is definitely I hack that I encountered when trying to implement delegation, hence the separate issue.

Thanks to this requirement it is ok from a security perspective, so I agree, let's leave it for this PR and reconsider it under #130 if need be.